batcher/aligned-batcher/src/lib.rs

```diff
@@ -950,13 +1028,26 @@ impl Batcher {
     /// Receives new block numbers, checks if conditions are met for submission and
     /// finalizes the batch.
     async fn handle_new_block(&self, block_number: u64) -> Result<(), BatcherError> {
-        let gas_price = match self.get_gas_price().await {
-            Some(price) => price,
-            None => {
-                error!("Failed to get gas price");
-                return Err(BatcherError::GasPriceError);
+        let gas_price_future = self.get_gas_price();
+        let disabled_verifiers_future = self.disabled_verifiers();
+
+        let (gas_price, disable_verifiers) =
+            tokio::join!(gas_price_future, disabled_verifiers_future);
+        let gas_price = gas_price.ok_or(BatcherError::GasPriceError)?;
+        let new_disable_verifiers =
+            disable_verifiers.map_err(|e| BatcherError::DisabledVerifiersError(e.to_string()))?;
+
+        let mut disabled_verifiers_lock = self.disabled_verifiers.lock().await;
+
+        if new_disable_verifiers != *disabled_verifiers_lock {
+            *disabled_verifiers_lock = new_disable_verifiers;
+            let has_disabled_verifiers = self
+                .queue_has_disabled_verifiers(new_disable_verifiers)
+                .await;
+            if has_disabled_verifiers {
+                self.flush_queue_and_clear_nonce_cache().await;
```

Comment:
Yeah, it could be a good first version for this. We can make sure that future versions checks for the proofs in the queue instead of flushing all of them everytime